sphinx/application.py

```diff
@@ -113,10 +112,10 @@
 class Sphinx(object):
 
     def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
-                 confoverrides=None, status=sys.stdout, warning=sys.stderr,
-                 freshenv=False, warningiserror=False, tags=None, verbosity=0,
-                 parallel=0):
-        # type: (unicode, unicode, unicode, unicode, unicode, Dict, IO, IO, bool, bool, unicode, int, int) -> None  # NOQA
+                 conffilename, confoverrides=None, status=sys.stdout,
```

Comment:
This brings incompatibility for initialization. It's not acceptable.
Could you give a default value to `conffilename`?

And more, could you move the parameter to the end of the argument list?
Some application which Sphinx uses as library depends on the order of arguments.
(ex.`sphinx/setup_command.py`, `tests/utils.py`)